drivers/display/display_sdl.c

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) 2018 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <display.h>
+
+#include <SDL.h>
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_DISPLAY_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(display_sdl);
+
+struct sdl_display_data {
+	SDL_Window *window;
+	SDL_Renderer *renderer;
+	SDL_Texture *texture;
+	bool display_on;
+	enum display_pixel_format current_pixel_format;
+	u8_t buf[4 * CONFIG_SDL_DISPLAY_X_RES * CONFIG_SDL_DISPLAY_Y_RES];
+};
+
+static struct sdl_display_data sdl_display_data;
+
+static int sdl_display_init(struct device *dev)
+{
+	struct sdl_display_data *disp_data =
+	    (struct sdl_display_data *)dev->driver_data;
+	LOG_DBG("Initializing display driver");
+
+	memset(disp_data, 0, sizeof(struct sdl_display_data));
+
+	disp_data->current_pixel_format = PIXEL_FORMAT_ARGB_8888;
+
+	disp_data->window =
+	    SDL_CreateWindow("Zephyr Display", SDL_WINDOWPOS_UNDEFINED,
+			     SDL_WINDOWPOS_UNDEFINED, CONFIG_SDL_DISPLAY_X_RES,
+			     CONFIG_SDL_DISPLAY_Y_RES, SDL_WINDOW_SHOWN);
+	if (disp_data->window == NULL) {
+		LOG_ERR("Failed to create SDL window: %s", SDL_GetError());
+		return -EIO;
+	}
+
+	disp_data->renderer =
+	    SDL_CreateRenderer(disp_data->window, -1, SDL_RENDERER_ACCELERATED);
```

Comment:
Note that you are never cleaning up on exit.
You can run it with `valgrind --leak-check=full zephyr/zephyr.exe` and will see there is ~5MB left allocated.
(At least I prefer not having (too much) left over to be sure I can still see real issues with memory leaks)

Reply:
Added exit functions ```sdl_display_cleanup``` and ```sdl_cleanup```, first function is cleaning up window, renderer and texture; the second function handles the SDL exit.

Even after calling all this cleanup functions still ~3MB is in use, some googling reveals this mainly due to X11 en OpenGL implementation.

Reply:
> Even after calling all this cleanup functions still ~3MB is in use,

Oh well.. I guess the SDL library is designed with a different mentality.

Reply:
That would be nice